.github/workflows/build.yml

```diff
@@ -43,56 +43,33 @@ jobs:
         uses: turtlebrowser/get-conan@main
 
-      - name: Setup Conan Packages
-        shell: bash
-        run: |
-          conan profile detect --force
-          conan install . -pr profiles/$OPERATING_SYSTEM/$BUILD_TYPE/$COMPILER --build=missing
-
       - name: Install CMake
         uses: lukka/get-cmake@latest
         
       - name: Configure
         shell: bash
         run: |
           if [ "${{ env.COMPILER }}" == "msvc" ]; then
-            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE=build/generators/conan_toolchain.cmake
+            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022"
+          elif [ "${{ env.COMPILER }}" == "clang-cl" ]; then
+            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022" -T ClangCL
           elif [ "${{ env.COMPILER }}" == "clang" ]; then
-            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_TOOLCHAIN_FILE=build/$BUILD_TYPE/generators/conan_toolchain.cmake
+            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_CXX_COMPILER=clang++
           elif [ "${{ env.COMPILER }}" == "gcc" ]; then
-            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_TOOLCHAIN_FILE=build/$BUILD_TYPE/generators/conan_toolchain.cmake
+            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_CXX_COMPILER=g++
           fi
 
       - name: Build
         run: cmake --build build --parallel --config ${{ env.BUILD_TYPE }}
 
-      - name: Test
+      - name: Unit Tests
         run: ctest --test-dir build --build-config ${{ env.BUILD_TYPE }} --output-on-failure
 
-      - name: Regression
+      - name: Regression Tests
         shell: bash
         run: |
           echo "Starting regression tests for $OPERATING_SYSTEM $BUILD_TYPE"
-          failures=()
-
-          for FORMAT in "ASCII" "ABBREV_ASCII" "BINARY" "FLATTENED_BINARY" "JSON"; do
-              ./build/bin/$BUILD_TYPE/converter_components database/messages_public.json regression/BESTUTMBIN.GPS $FORMAT > /dev/null
-
-              if ! cmp -s regression/BESTUTMBIN.GPS.$FORMAT regression/targets/BESTUTMBIN.GPS.$FORMAT; then
-                  failures+=("OEM-$FORMAT")
-              fi
-              
-              if ! cmp -s regression/BESTUTMBIN.GPS.$FORMAT.UNKNOWN regression/targets/BESTUTMBIN.GPS.$FORMAT.UNKNOWN; then
-                  failures+=("OEM-$FORMAT.UNKNOWN")
-              fi
-          done
-          
-          if [ ${#failures[@]} -gt 0 ]; then
-              echo "One or more regression tests failed! ${failures[@]}"
-              echo "See the README if you don't know how to fix this."
-              exit 1
-          fi
-          
+          bash regression/run_tests.sh ./build/bin/$BUILD_TYPE/converter_components
```

Comment:
So is the advantage of moving the regression testing script to a shell script that we can run it locally? If so, the README should be updated to reflect this. I didnt notice any functional changes otherwise.

Reply:
Indeed, that was the intention. Makes it easier to test the changes to CI locally, before pushing.